pay_android/android/src/main/kotlin/io/flutter/plugins/pay_android/GooglePayHandler.kt

```diff
@@ -228,12 +234,16 @@ class GooglePayHandler(private val activity: Activity) :
      * At this stage, the user has already seen a popup informing them an error occurred.
      * Normally, only logging is required.
      *
-     * @param statusCode the value of any constant from [CommonStatusCodes] or one of the
+     * @param statusCodeStr the value of any constant from [CommonStatusCodes] or one of the
      * []WalletConstants].ERROR_CODE_* constants.
+     * @param errorMessage the reason for the error.
      *
      * @see [
      * Wallet constants library](https://developers.google.com/android/reference/com/google/android/gms/wallet/WalletConstants.constant-summary)
      */
-    private fun handleError(statusCode: Int) =
-            loadPaymentDataResult.error(statusCode.toString(), "", null)
+    private fun handleError(statusCodeStr: String, errorMessage: String = "") {
+        if (isResponseSent) return@handleError
+        loadPaymentDataResult.error(statusCodeStr.toString(), errorMessage, null)
+        isResponseSent = true
+    }
```

Comment:
```suggestion
    private fun handleError(statusCode: String, errorMessage: String = "") {
    if (isResponseSent) return

    hasResponseBeenSent = true
    loadPaymentDataResult.error(statusCode, errorMessage, null)
}

```

We can simplify the return statement: removing the unnecessary @handleError label in the return statement, as it’s redundant here.